eth_typing/abi.py

```diff
@@ -58,7 +58,7 @@ class ABIEvent(TypedDict):
     """Event ABI type."""
     anonymous: NotRequired[bool]
     """If True, event is anonymous. Cannot filter the event by name."""
-    inputs: NotRequired[Sequence["ABIComponent"]]
+    inputs: NotRequired[Sequence[Union["ABIComponent", "ABIComponentIndexed"]]]
```

Comment:
I'm questioning if this is event necessary, `ABIComponentIndexed` extends `ABIComponent`. So as long as you specify the type when using an `"indexed"` component, typing works fine.

```python
ABIEvent(
                {
                    "type": "event",
                    "name": "LogSingleWithIndex",
                    "inputs": [
                        ABIComponentIndexed(
                            {"name": "arg0", "type": "uint256", "indexed": True}
                        )
                    ],
                }
            )
```

Reply:
That makes sense to me. indexed exists to narrow the type down when it makes sense to. We should only need to support `ABIComponent`.